test/utils/toolpadApi.ts

```diff
@@ -0,0 +1,40 @@
+import './fetchPolyfill';
+import { createRpcClient } from '../../packages/toolpad-app/src/rpcClient';
+import type { ServerDefinition } from '../../packages/toolpad-app/pages/api/rpc';
+import generateId from './generateId';
+import type { CreateAppOptions } from '../../packages/toolpad-app/src/server/data';
+
+export function getApi(endpoint: string) {
+  return createRpcClient<ServerDefinition>(endpoint);
+}
+
+interface CreateApplicationParams {
+  name?: string;
+  appTemplateId?: string;
+  dom?: any;
+}
+
+export async function createApplication(
+  baseUrl: string,
+  { name = `App ${generateId()}`, appTemplateId, dom }: CreateApplicationParams = {},
+) {
+  const api = getApi(new URL('/api/rpc', baseUrl).href);
```

Comment:
Could this `api` variable not be global if the playwright `baseUrl` is always the same?
Just considering if it's possible in case `baseUrl` could be retrieved outside the tests, but I don't have the whole context.

Reply:
Can't make it global because the `baseURL` is not available globally unless we stop relying on the config as single source of truth. I reworked it though to make `api` available inside the fixture.